google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/AbstractSubscriberConnection.java

```diff
@@ -53,30 +44,30 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
-import javax.annotation.Nullable;
 import org.joda.time.Duration;
 import org.joda.time.Instant;
 import org.joda.time.Interval;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Implementation of {@link Subscriber}. */
-final class SubscriberConnection extends AbstractService {
-  private static final Logger logger = LoggerFactory.getLogger(SubscriberConnection.class);
+/**
+ * Abstract base implementation class of a subscriber connection in charge of receiving subscription
+ * messages.
+ */
+abstract class AbstractSubscriberConnection extends AbstractService {
```

Comment:
I would much prefer a containment model over an inheritance model - it would be much easier to understand the interactions between the classes.